lustre-collector/src/recovery_status_parser.rs

```diff
@@ -113,22 +131,22 @@ where
     many(choice((
         status_line()
             .skip(optional(newline()))
-            .map(RecoveryStat::Status)
-            .map(Some),
-        clients_line("completed_clients")
-            .skip(optional(newline()))
-            .map(RecoveryStat::Completed)
-            .map(Some),
+            .map(|x| vec![RecoveryStat::Status(x)]),
+        simple_client_stat("recovery_duration", RecoveryStat::RecoveryDuration),
+        simple_client_stat("completed_clients", RecoveryStat::Completed),
+        simple_client_stat("time_remaining", RecoveryStat::TimeRemaining),
+        simple_client_stat("evicted_clients", RecoveryStat::Evicted),
         clients_line("connected_clients")
             .skip(optional(newline()))
-            .map(RecoveryStat::Connected)
-            .map(Some),
-        clients_line("evicted_clients")
-            .skip(optional(newline()))
-            .map(RecoveryStat::Evicted)
-            .map(Some),
+            .map(|(x, y)| {
+                let mut stats = vec![RecoveryStat::Connected(x)];
+                if let Some(total) = y {
+                    stats.push(RecoveryStat::Total(Some(total)));
+                }
+                stats
```

Comment:
Stylistic opportunity - I think this would look better with an immutable approach:
```
iter::once(RecoveryStat::Connected(x))
    .chain(y.map(|total| vec![RecoveryStat::Total(Some(total))]).unwrap_or_default())
    .collect()
```

If you want to keep the mutable version let's at least add some newlines between the statements.